dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/datavalue/DataValueServiceTest.java

```diff
@@ -104,27 +112,35 @@ class DataValueServiceTest extends TransactionalIntegrationTest
     public void setUpTest()
         throws Exception
     {
-        deA = createDataElement( 'A' );
-        deB = createDataElement( 'B' );
-        deC = createDataElement( 'C' );
-        deD = createDataElement( 'D' );
-        dataElementService.addDataElement( deA );
-        dataElementService.addDataElement( deB );
-        dataElementService.addDataElement( deC );
-        dataElementService.addDataElement( deD );
-        peA = createPeriod( getDay( 5 ), getDay( 6 ) );
-        peB = createPeriod( getDay( 6 ), getDay( 7 ) );
-        peC = createPeriod( getDay( 7 ), getDay( 8 ) );
-        peX = createPeriod( getDay( 27 ), getDay( 28 ) );
-        ouA = createOrganisationUnit( 'A' );
-        ouB = createOrganisationUnit( 'B' );
-        ouC = createOrganisationUnit( 'C' );
-        ouD = createOrganisationUnit( 'D' );
-        organisationUnitService.addOrganisationUnit( ouA );
-        organisationUnitService.addOrganisationUnit( ouB );
-        organisationUnitService.addOrganisationUnit( ouC );
-        organisationUnitService.addOrganisationUnit( ouD );
+        setup.addDataElement( "A", withDefaultSetup );
+        setup.addDataElement( "B", withDefaultSetup );
+        setup.addDataElement( "C", withDefaultSetup );
+        setup.addDataElement( "D", withDefaultSetup );
+        setup.addPeriod( "2022-09-05", withDefaultSetup );
+        setup.addPeriod( "2022-09-06", withDefaultSetup );
+        setup.addPeriod( "2022-09-07", withDefaultSetup );
+        setup.addPeriod( "2022-09-27", withDefaultSetup );
+        setup.addOrganisationUnit( "A", withDefaultSetup );
+        setup.addOrganisationUnit( "B", withDefaultSetup );
+        setup.addOrganisationUnit( "C", withDefaultSetup );
+        setup.addOrganisationUnit( "D", withDefaultSetup );
```

Comment:
Introduce a new method overload to avoid specifying `withDefaultSetup` everywhere?